creator-node/test/fileManager.js

```diff
@@ -29,124 +31,247 @@ const req = {
+
+      try {
+        await saveFileToIPFSFromFS(req, srcPath, fileType, sourceFile)
+        assert.fail('Should not have passed if ipfs is down.')
+      } catch (e) {
+        assert.deepStrictEqual(e.message, 'ipfs is down!')
+      }
+    })
+
+    /**
+     * Given: a file is being saved to ipfs from fs
+     * When: file syncing fails
+     * Then: an error is thrown
+     */
+    it('should throw an error if file syncing fails', async () => {
+      sinon.stub(fs, 'copyFileSync').throws(new Error('Failed to sync over files!!'))
+
+      try {
+        await saveFileToIPFSFromFS(req, srcPath, fileType, sourceFile)
+        assert.fail('Should not have passed if file syncing fails.')
+      } catch (e) {
+        assert.deepStrictEqual(e.message, 'Failed to sync over files!!')
+      }
+    })
+
+    /**
+     * Given: a file is being saved to ipfs from fs
+     * When: the db connection is down
+     * Then: an error is thrown
+     */
+    it('should throw an error if db connection is down', async () => {
+      sinon.stub(models.File, 'findOrCreate').rejects(new Error('Failed to find or create file!!!'))
+
+      try {
+        await saveFileToIPFSFromFS(req, srcPath, fileType, sourceFile)
+        assert.fail('Should not have if db connection is down.')
+      } catch (e) {
+        assert.deepStrictEqual(e.message, 'Failed to find or create file!!!')
+      }
+    })
+
+    /**
+     * Given: a file is being saved to ipfs from fs
+     * When: everything works as expected
+     * Then:
+     *  - 1 segment should be saved in <storagePath>/QmSMQGu2vrE6UwXiZDCxyJwTsCcpPrYNBPJBL4by4LKukd
+     *  - that segment content should match the original sourcefile
+     *  - that segment should be present in IPFS
+     */
+    it('should pass saving file to ipfs from fs (happy path)', async () => {
+      sinon.stub(models.File, 'findOrCreate').returns([{ dataValues: 'data' }])
+
+      try {
+        await saveFileToIPFSFromFS(req, srcPath, fileType, sourceFile)
+      } catch (e) {
+        assert.fail(e.message)
+      }
+
+      // 1 segment should be saved in <storagePath>/QmSMQGu2vrE6UwXiZDCxyJwTsCcpPrYNBPJBL4by4LKukd
+      const segmentCID = 'QmSMQGu2vrE6UwXiZDCxyJwTsCcpPrYNBPJBL4by4LKukd'
+      const syncedSegmentPath = path.join(storagePath, segmentCID)
+      assert.ok(fs.existsSync(syncedSegmentPath))
+
+      // the segment content should match the original sourcefile
+      const syncedSegmentBuf = fs.readFileSync(syncedSegmentPath)
+      const originalSegmentBuf = fs.readFileSync(srcPath)
+      assert.deepStrictEqual(originalSegmentBuf.compare(syncedSegmentBuf), 0)
+
+      // the segment should be present in IPFS
+      let ipfsResp
+      try {
+        ipfsResp = await ipfs.cat(segmentCID)
+      } catch (e) {
+        // If CID is not present, will throw timeout error
+        assert.fail(e.message)
+      }
+
+      assert.deepStrictEqual(originalSegmentBuf.compare(ipfsResp), 0)
```

Comment:
😍 this is it. the good stuff. hahaha. any reason you didn't add this check to other tests?

Reply:
:') bc my brain farted. added!